bin/execute_and_publish_performance_test.sh

```diff
@@ -1,13 +1,24 @@
 #!/bin/bash
 
+# Setup
 current_time=$(date "+%Y.%m.%d-%H.%M.%S")
 perf_test_aws_s3_bucket=${PERF_TEST_AWS_S3_BUCKET:-notify-performance-test-results-staging}
 perf_test_csv_directory_path=${PERF_TEST_CSV_DIRECTORY_PATH:-/tmp/notify_performance_test}
-
 mkdir -p $perf_test_csv_directory_path/$current_time
 
+# Run old performance test and copy results to S3
 locust --headless --config tests-perf/locust/locust.conf --html $perf_test_csv_directory_path/$current_time/index.html --csv $perf_test_csv_directory_path/$current_time/perf_test
-
 aws s3 cp $perf_test_csv_directory_path/ "s3://$perf_test_aws_s3_bucket" --recursive || exit 1
 
+# Sleep 15 minutes to allow the system to stabilize
+sleep 900
+
+# Run email send rate performance test
+# This configuration should send 10K emails / minute for 10 minutes for 100K emails total.
+# We run this test on Tuesday through Friday (just after midnight UTC) only.
+if [ "$(date +%u)" -ge 2 ] && [ "$(date +%u)" -le 5 ]; then
+    locust --headless --host https://api.staging.notification.cdssandbox.xyz --locustfile tests-perf/locust/send_rate_email.py --users 5 --run-time 10m --spawn-rate 1
```

Comment:
Why is the locust test named send_rate_email? is this to send to a specific rate? or to hit the rate limit?

Reply:
We are not copying results into s3 for these?

Reply:
We could but the main point in this test is just to see how high we get the send rate, which isn't going to be in the locust results.

Reply:
We habve the blazer query but would be easy to add it to QuickSight too

Reply:
Just saw your first question! I named it that since all it's testing is the max send rate we can achieve. Ie it does bulk sends to get the system running at max capacity. After the test we can look at a query / dashboard to see what send rate was reached.